techTree.md

```diff
@@ -0,0 +1,97 @@
+# Stateless Tech Tree
+
+A 'roadmap' delineates centrally determined deliverables and milestones. Because the Stateless Ethereum effort is not centrally planned, it's helpful to instead use a [tech tree](https://en.wikipedia.org/wiki/Technology_tree) to organize and record ongoing work in the Stateless Ethereum initiative. This helps to map out a shared understanding of 'the critical path' and dependencies in research / implementation.
+
+![tech_tree](./assets/Stateless_Tech_Tree_06-2020.png)
+
+> ### Key
+>
+> * **Purple**: Specific pre-requisites for a major milestone. Any topic that could or will eventually be refined into an EIP, as well as anything considered on 'the critical path'. 
+> * **Yellow**: Support work needed for a major milestone, such as software tools or data analysis.
+> * **Green**: Nice-to-haves that are not critical but still valuable to Stateless Ethereum and/or Eth2. 
+
+
+
+## Witness Prototype
+
+It's possible to build a stateless client today with the current network primitives; beam sync is an example of a semi-stateless client implementation. The objective of this milestone is to have implemented clients on mainnet passing around prototype witnesses. It's likely that the first witnesses will be too large to propagate efficiently, but a first implementation will be a good starting point to improve upon through other objectives such as code merkleization and a binary state trie. 
+
+The starting point for a witness format prototype is the [Witness Specification](./witness.md), which aims to define witnesses without ambiguity for the sake of soundness in implementation. 
+
+The specification is still in draft form and contributions, especially in writing tests, are welcome. 
+
+In addition to the specification, there are two noteworthy issues that a witness implementation must 'solve' in some way.
+
+#### Witness Indexing
+
+We need a reliable way of determining which witness corresponds to which block and associated state. This could be as simple as putting a `witnessHash` field into the block header, or something else that serves the same purpose but in a different way.
+
+#### Stateless tx Validation
+
+This is an interesting early problem [thoroughly summarized on the ethresearch forums](https://ethresear.ch/t/the-current-transaction-validation-rules-require-access-to-the-account-trie/7046). In summary, clients need to quickly check if incoming transactions (waiting to be mined into a future block) are at least *eligible* to be included in a future block. 
+
+
+
+## EVM
+
+Stateless Ethereum will require a change to EVM semantics, particularly in gas execution and opcode repricing.  Because the EVM is currently an implicit (and not explicit) Ethereum State Transition Function, changes to semantics can be quite complex, and will likely affect future work on Eth1x as well as Eth2.
```

Comment:
What is the difference between an implicit and explicit state transition function?

Reply:
Ah, I borrowed this phrase from Danny. Meaning that the EVM lacks an explicitly defined state transition function like would be needed for Execution Environments. Rather, it's just the action of transitioning state according to the protocol that defines valid vs invalid state transition. The EVM is the function.

Reply:
The words "implicit" vs. "explicit" remain unclear to me. The yellowpaper defines a state transition function $\Upsilon$ which maps a prestate and transaction to the poststate. (I suspect "implicit" vs "explicit" has to do with the state transition function taking the full state as input vs. taking a witness as input.)